### PR TITLE
[Feature] TVR infra — LocalMetastore getCurrentTvrSnapshot + listTableDeltaTraits

### LINKED ISSUE #73115: [Feature] Implement TVR primitives in LocalMetastore for native storage
# [Feature] Implement TVR primitives in LocalMetastore for native storage

Sub-issue of #67632

## Goal

Implement `getCurrentTvrSnapshot` and `listTableDeltaTraits` for `LocalMetastore` to support incremental change sets for native StarRocks tables (OLAP + CLOUD_NATIVE).

## Background

`IcebergMetadata` and `PaimonMetadata` implement these two TVR methods. `LocalMetastore` inherits the noop defaults that return `TvrTableSnapshot.empty()`. `MVIVMRefreshProcessor.getBaseTableChangedVersionRange` (lines 198-270) cannot compute deltas for native tables.

## Proposed change

```java
@Override
public TvrTableSnapshot getCurrentTvrSnapshot(String dbName, Table table) {
    // Return max(physicalPartition.getVisibleVersion()) wrapped as a TvrTableSnapshot.
}

@Override
public List<TvrTableDeltaTrait> listTableDeltaTraits(
        String dbName, Table table,
        TvrTableSnapshot fromExclusive, TvrTableSnapshot toInclusive) {
    // - equal from/to → empty list (matches IcebergMetadata:786).
    // - partition-shape change → throw StarRocksConnectorException("snapshot ancestry broken...");
    //   caller rewraps via MVIVMRefreshProcessor.formatPartitionShapeChangeError (lines 277-284).
    // - otherwise → emit one TvrTableDeltaTrait classified by KeysType (see Q1).
}
```

Scoped to `OLAP` and `CLOUD_NATIVE` with identical implementations using `PhysicalPartition.getVisibleVersion`. No modifications to `IVMAnalyzer.SUPPORTED_TABLE_TYPES`; this is a separate follow-up.

## Open questions

1. Per-version op-type classification
`TransactionType` has only `TXN_NORMAL` and `TXN_REPLICATION`. There is no `INSERT/UPSERT/DELETE` distinction. `PhysicalPartition.versionTxnType` reflects only the current version.

| Option | Approach | Trade-off |
|---|---|---|
| **A (proposed)** | Classify by `KeysType` alone: DUP → MONOTONIC; PK/UK/AGG → RETRACTABLE | Conservative — append-only PK loads incur PCT fallback. Zero new infra. |
| B | Add a per-version `(version, opType)` log on `PhysicalPartition`, populated via the txn log applier | Schema change on a hot path; precise classification. |
| C | Read TxnLog from BE at query time | Precise but `TransactionState` is purged after retention. |

2. Snapshot id for partitioned tables
Iceberg uses a single snapshot id. Native tables have per-partition `visibleVersion` which can diverge if loads target different partitions.

| Option | Approach | Trade-off |
|---|---|---|
| **A (proposed)** | `max(visibleVersion)` — single Long, matches `TvrTableSnapshot`'s shape | Refresh-replan may be over-pessimistic when partitions advance independently. |
| B | Per-partition tuple `Map<PartitionId, Long>` | Wire-format change to `TvrTableSnapshot`; impacts `MVIVMRefreshProcessor` and `IcebergMetadata` callers. |

## Acceptance criteria

* `getCurrentTvrSnapshot` returns a non-empty snapshot equal to `max(visibleVersion)` for native tables.
* `listTableDeltaTraits` emits one MONOTONIC trait for DUP_KEYS, one RETRACTABLE trait for PK/UK/AGG_KEYS (per Q1 Option A).
* Equal `from`/`to` returns empty list
* Partition-shape change throws `StarRocksConnectorException` of the shape caught by `MVIVMRefreshProcessor.isAncestryBrokenError`
* New unit test covers the four `KeysType` values + edge cases
* Existing `IVMBasedMvRefreshProcessorIcebergTest` still passes

## Depends on

#69531 — Fix TVR watermark advancing in PCT fallback. Without this fix, the RETRACTABLE deltas produced by Q1 Option A loop between IVM attempt and PCT fallback on every PK/UK/AGG load.

### PR DESCRIPTION
**Pull request by jayakasadev**
## Why I'm doing:

`LocalMetastore` had no TVR (Time-Varying Relation) implementation, so native IVM could not compute change sets for OLAP/CLOUD_NATIVE tables and `MVIVMRefreshProcessor` made no progress.

## What I'm doing:

Adds two `ConnectorMetadata` overrides to `LocalMetastore`:

- **`getCurrentTvrSnapshot`** — returns the **max** `versionEpoch` across physical partitions as the change watermark. Max advances on any load and cannot overflow (summing epochs overflowed signed `long` past ~22 partitions).
- **`listTableDeltaTraits`** — emits one RETRACTABLE delta per refresh (DELETE is legal on every key type, so native deltas cannot be proven append-only), and throws a typed `TvrAncestryBrokenException` when the watermark regresses.

`IVMAnalyzer.SUPPORTED_TABLE_TYPES` still gates on `{ICEBERG}` only; unlocking CLOUD_NATIVE is Phase 2.

**Depends on #76567** — the watermark only moves once `versionEpoch` advances on every commit. Closes #73115

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
  - [ ] This pr needs auto generate documentation
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 4.1
  - [ ] 4.0
  - [ ] 3.5

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by CelerData-Reviewer** [comment]
@codex review

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -5755,4 +5758,52 @@ public void alterTableAutoIncrement(String dbName, String tableName, long newAut
...
+
+    @Override
+    public TvrTableSnapshot getCurrentTvrSnapshot(String dbName, Table table) {
+        if (!(table instanceof OlapTable)) {
+            return TvrTableSnapshot.empty();
+        }
+        OlapTable olapTable = (OlapTable) table;
+        long maxVersion = olapTable.getAllPhysicalPartitions().stream()
+                .mapToLong(PhysicalPartition::getVisibleVersion)
+                .max()
```
[the remark this answers is not in the document]
Comment (jayakasadev) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java]: Fixed in 8d9a93c. Switched to `sum(visibleVersion)` across all partitions — any partition increment now advances the watermark, so loads into non-max partitions are no longer missed.

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -5755,4 +5758,52 @@ public void alterTableAutoIncrement(String dbName, String tableName, long newAut
...
+                ? Optional.empty()
+                : Optional.of(fromSnapshotExclusive.getSnapshotId());
+        TvrTableDelta delta = TvrTableDelta.of(fromOpt, Optional.of(toVersion));
+        return Collections.singletonList(classifyNativeDelta(olapTable, delta));
+    }
+
+    private static TvrTableDeltaTrait classifyNativeDelta(OlapTable table, TvrTableDelta delta) {
+        return table.getKeysType() == KeysType.DUP_KEYS
+                ? TvrTableDeltaTrait.ofMonotonic(delta)
+                : TvrTableDeltaTrait.ofRetractable(delta);
```
[the remark this answers is not in the document]
Comment (jayakasadev) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java]: Fixed in 8d9a93c. Dropped the DUP_KEYS=MONOTONIC branch — `classifyNativeDelta` now always returns RETRACTABLE. DUP_KEYS supports DELETE predicates so append-only can't be guaranteed; the conservative choice is correct for v1. A comment on the method notes the path to MONOTONIC promotion once per-version op-type tracking exists.

**Comment by CelerData-Reviewer** [comment]
@codex review

**Review by jaya-traversal — APPROVED** [review]

**Comment by CelerData-Reviewer** [comment]
@codex review

**Comment by jayakasadev** [comment]
@Youngwb checking in to see if any other changes are needed here

**Comment by CelerData-Reviewer** [comment]
@codex review
